### PR TITLE
pink: Add API ecsda_sign(verify)_prehashed

### PR DESCRIPTION
**Pull request by kvinwang**
And some refactoring to the signing extension parameters.

### CODE REVIEW

File: crates/pink/src/runtime/extension.rs [hunk 1]
```diff
@@ -323,4 +357,22 @@ impl PinkExtBackend for CallInCommand {
     fn is_running_in_command(&self) -> Result<bool, Self::Error> {
         Ok(true)
     }
+
+    fn ecdsa_sign_prehashed(
```
Comment (h4x3rotab) [inline crates/pink/src/runtime/extension.rs:361]: Why do we offer prehash signing only for ecdsa?

Comment (kvinwang) [reply crates/pink/src/runtime/extension.rs:361]: The other two don't call a hash before passing the payload into their inner sign().

Comment (h4x3rotab) [reply crates/pink/src/runtime/extension.rs:361]: Could we double check? Usually they should accept 256 bits pre-hashed message, but apparently they accept a `Vec<u8>` instead. So I guess some hash is still used.

Comment (kvinwang) [reply crates/pink/src/runtime/extension.rs:361]: Double checked.

**Pull request merged**